app/lib/languages.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+# Helper methods for languages
+module Languages
+  def self.supported_language?(lang)
+    return false if lang.nil?
+
+    # If the supported field exists at all for a language, then
+    # that language can be considered supported
+    all_languages&.dig(lang.to_sym, :supported)
+  end
+
+  # Even though some languages may be supported, we are unable to send
+  # voice-calls in that language (typically due to Twilio limitations)
+  def self.voice_unsupported?(lang)
+    return false if lang.nil?
+
+    !all_languages&.dig(lang.to_sym, :supported, :phone)
+  end
+
+  def self.all_languages
+    PATIENT_HELPER_FILES[:languages]
+  end
+
+  def self.attempt_language_matching(lang)
+    # This function returns the `lang` passed in if it can't be matched
+    # Else returns the matched 'lang' iso code (stringified)
+    matched_val = normalize_and_get_language_name(lang)
+    matched_val ? matched_val.to_s : lang
+  end
+
+  # This function will attempt to match the input to a language in the system
+  # PARAM: `lang` can be a three-letter iso-639-2t code, a two-letter iso-639-1 code, or the name (not case sensitive)
+  # PARAM EXAMPLES: 'eng', 'en', 'English', 'ENGLISH' <-- All will map to 'eng'
+  # RETURN VALUE: `nil` if unmatchable, else the three-letter iso code ('eng')
+  def self.normalize_and_get_language_name(lang)
+    return nil if lang.nil?
+
+    lang = lang.to_s.downcase.strip
+    matched_language = nil
+    matched_language = lang.to_sym if all_languages[lang.to_sym].present?
+    return matched_language unless matched_language.nil?
+
+    matched_language = all_languages.find { |_key, val| val[:display]&.casecmp(lang)&.zero? }
```

Comment:
@holmesie @Bialogs since this could potentially be used for every patient in the import loop, would it be valuable to define an inverted lookup hash for this as something like:
```
Languages.all_languages.stringify_keys.transform_values { |v| v[:display]&.downcase }.invert
```
that way these lookups can be O(1) instead of O(n)?

Reply:
same for `iso6391code`

Reply:
We're currently refactoring this to not be O(n)